### PR TITLE
Fixes #661 Data now retains in Control Fragment Read

### LINKED ISSUE #661: Not saved the changes of main control in Control Application
**Actual Behaviour**
When increment or decrement the PV values and set it and then go to read section then these values of PV are not saved.

**Expected Behaviour**
When going from main control.to other sections, the changes made in main control should be saved

**Steps to reproduce it**

whatsapp video 2017-10-06 at 22 51 26

**Would you like to work on the issue?**
yes.

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes issue #661 

Changes: I have created a ControlActivityCommon class which stores the values of edit text in the fragment when the set button is clicked so that when fragment is restored it check the values of common class variable otherwise the default values are shown.
These common values will remain until the activity is not  destroyed.

All the mentors plz review.

ezgif com-video-to-gif

### CODE REVIEW

**Comment by CloudyPadmal** [comment]
Thank you for the PR! Looks good @Avjeet ! Can you send the APK to test it?

File: app/src/main/java/org/fossasia/pslab/fragment/ControlFragmentMain.java [hunk 1]
```diff
@@ -19,25 +19,30 @@
...
+    private ControlMainAdapter mAdapter;
+
     public static ControlFragmentMain newInstance() {
         return new ControlFragmentMain();
     }
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+         mAdapter = new ControlMainAdapter(new String[]{"PV1", "PV2", "PV3", "PCS", "WAVE 1" , "WAVE 2" , "SQUARE"});
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/ControlFragmentMain.java]: There is an extra space here at the beginning of the line

File: app/src/main/java/org/fossasia/pslab/others/ScienceLabCommon.java [hunk 2]
```diff
@@ -35,4 +35,6 @@ public static ScienceLabCommon getInstance() {
         }
         return scienceLabCommon;
     }
+
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/others/ScienceLabCommon.java]: These new lines are not needed!

**Comment by Avjeet** [comment]
where should i send the apk?
Should i upload it here only.

**Comment by akarshan96** [comment]
@Avjeet upload the apk here.

**Comment by Avjeet** [comment]
[app-debug.zip](https://github.com/fossasia/pslab-android/files/1732462/app-debug.zip)

here is the apk zip file.

**Comment by Avjeet** [comment]
one heads up the data only retains if the user press the set button unless it will return to default.

**Comment by CloudyPadmal** [comment]
Let me review! 👍

**Comment by CloudyPadmal** [comment]
@Avjeet When I try to set three values for PV1, PV2 and PV3 and move to another tab, PV3 value reset back to 0.00 and that value is set to PV2 replacing the old value of PV2. Please fully test the functionality! 👍

**Comment by Avjeet** [comment]
@CloudyPadmal that was minor bug which is removed ,here is the new apk plz review it.
[app-debug2.zip](https://github.com/fossasia/pslab-android/files/1734502/app-debug2.zip)

**Comment by asitava1998** [comment]
@Avjeet Next time onwards squash all commits in to one commit

**Pull request merged**